src/simulators/tensor_network/svd.cpp

```diff
@@ -530,30 +487,34 @@ status csvd(cmatrix_t &A, cmatrix_t &U,rvector_t &S,cmatrix_t &V)
 
 void csvd_wrapper (cmatrix_t &A, cmatrix_t &U,rvector_t &S,cmatrix_t &V)
 {
-	cmatrix_t coppied_A = A;
-	int times = 0;
-	if(DEBUG) cout << "1st try" << endl;
-	status current_status = csvd(A, U, S, V);
-	if (current_status == SUCCESS)
-	  return;
-	if(DEBUG) cout << "1st try success" << endl;
-
-	while(times <= NUM_SVD_TRIES && current_status == FAILURE)
-	  {
-	    times++;
-	    coppied_A = coppied_A*mul_factor;
-	    A = coppied_A;
-	    if(DEBUG) cout << "another try" << endl;
-	    current_status = csvd(A, U, S, V);
-	  }
-	if(times == NUM_SVD_TRIES) {
-	  cout << "SVD failed" <<endl;
-	  assert(false);
-	}
+  cmatrix_t copied_A = A;
+  int times = 0;
+#ifdef DEBUG
+  cout << "1st try" << endl;
+#endif
+  status current_status = csvd(A, U, S, V);
+  if (current_status == SUCCESS)
+    return;
+
+  while(times <= NUM_SVD_TRIES && current_status == FAILURE)
+    {
+      times++;
+      copied_A = copied_A*mul_factor;
+      A = copied_A;
 
-	//Divide by mul_factor every singular value after we multiplied matrix a
-	for(int k = 0; k < S.size(); k++)
-			S[k] /= pow(mul_factor, times);
+#ifdef DEBUG
+      cout << "SVD trial #" << times << endl;
+#endif
+
+      current_status = csvd(A, U, S, V);
+    }
+  if(times == NUM_SVD_TRIES) {
+    std::cerr << "SVD failed" << std::endl;
```

Comment:
We don't use standard output/error to write "logs" or "traces". I'd enclose this message with `#ifdef DEBUG/#endif` directives.

Reply:
This is also a real error. If we get this error, the algorithm will not continue correctly., so it is better to exit. I changed to a "throw". Let me know if you think this should be handled differently.